src/lib/marshal/message-type/reader.ts

```diff
@@ -0,0 +1,224 @@
+import path from "node:path";
```

Comment:
reader.ts and writer.ts are copied over from other modules and adjusted for message types, mostly the same as how email layouts are handled, for exmaple.